contrib/migration_script.py

```diff
@@ -0,0 +1,170 @@
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Migrate jobs to new namespace'
+    )
+    parser.add_argument(
+        '--server',
+        required=True,
+        # default='http://tronplayground-uswest1adevc.dev.yelpcorp.com:8089',
+        help='specify the location of tron master'
+    )
+    parser.add_argument(
+        '--old-ns',
+        required=True,
+        # default='dsc-test-service',
+        help='Old namespace'
+    )
+    parser.add_argument(
+        '--new-ns',
+        required=True,
+        # default='test_namespace',
+        help='New namespace'
+    )
+    parser.add_argument(
+        'source',
+        help='source file to get list of jobs'
+    )
+    args = parser.parse_args()
+    return args
+
+
+def check_job_if_running(jobs_status, job_name):
+    for job_status in jobs_status:
+        if job_status['name'] == job_name:
+            status = job_status['status']
+            if status == 'running':
+                print(bcolors.FAIL + 'job {} is still running, can not migrate'.format(job_name) + bcolors.ENDC)
+                return False
+            else:
+                print(bcolors.OKGREEN + 'job {} is not running, can migrate'.format(job_name) + bcolors.ENDC)
+                return True
+
+    print(bcolors.FAIL + 'Can not find the job {}'.format(job_name) + bcolors.ENDC)
+    return False
+
+
+def command_jobs(command, jobs, args):
+    data = {'command': command}
+    command_flag = True
+    for job in jobs:
+        if command == 'enable':
+            job_name = args.new_ns + '.' + job['name']
+        else:
+            job_name = args.old_ns + '.' + job['name']
+
+        if command == 'move':
+            data = {'command': command, 'old_name': args.old_ns + '.' + job['name'], 'new_name': args.new_ns + '.' + job['name']}
+            uri = urljoin(args.server, 'api/jobs')
+        else:
+            data = {'command': command}
+            uri = urljoin(args.server, 'api/jobs/' + job_name)
+
+        response = client.request(uri, data=data)
+        if response.error:
+            print(bcolors.FAIL + 'Failed to {} {}'.format(command, job_name) + bcolors.ENDC)
+            command_flag = False
+        else:
+            print(bcolors.OKGREEN + 'Succeed to {} {}'.format(command, job_name) + bcolors.ENDC)
+    return command_flag
+
+
+def ssh_command(hostname, command):
+    ssh = subprocess.Popen(["ssh", "-A", hostname, command], shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    result = ssh.stdout.readlines()
+    error = ssh.stderr.readlines()
+    if error != []:
+        print(bcolors.BOLD + 'Execute command {}: {}'.format(command, error) + bcolors.ENDC)
+    return result
+
+
+def main():
+    args = parse_args()
+    filename = args.source
+    hostname = urlparse(args.server).hostname
+    if filename.endswith(".yaml"):
```

Comment:
For this bulk migration, working at a whole file at a time is good and efficient. In the long term we will probably need a more fine-grained way to do this (per-job) because MASTER will basically never been ready to move (always some job will be running probably)